sirepo/package_data/static/js/activait.js

```diff
@@ -1120,6 +1120,74 @@ SIREPO.app.directive('columnSelector', function(appState, mlService, panelState,
     };
 });
 
+
+SIREPO.app.directive('imagePreviewPanel', function(requestSender) {
+    return {
+        restrict: 'A',
+        scope: {},
+        template: `
+        <div>
+          <img class="img-responsive srw-processed-image" />
+          <div data-ng-if="isLoading()" class="progress">
+            <div class="progress-bar progress-bar-striped active" role="progressbar" aria-valuenow="{{ simState.getPercentComplete() }}" aria-valuemin="0" aria-valuemax="100" data-ng-attr-style="width: {{ simState.getPercentComplete() || 100 }}%"></div>
+          </div>
+          <div style="display: flex; justify-content: space-between;" data-ng-if="! isLoading()">
+            <button data-ng-click="prev()"> < prev image set </button>
+            <button data-ng-click="next()"> next image set > </button>
+          </div>
+        </div>
+        `,
+        controller: function($scope, appState) {
+            var idx = 0;
+            var uris;
+            var loading = true;
+
+            $scope.isLoading = () => loading;
+
+            $scope.updateUriIndex = function(increment){
+                if ((0 <= idx + increment) && (idx + increment <= 4)) {
+                    idx = idx + increment;
+                    setImageFromUriIndex(idx);
+                }
+            };
+
+            $scope.prev = function() {
+                $scope.updateUriIndex(-1);
+            };
+
+            $scope.next = function() {
+                $scope.updateUriIndex(1);
+            };
+
+            const setImageFromUriIndex = (index) => {
+                if ($('.srw-processed-image').length) {
+                    $('.srw-processed-image')[0].src = uris[index];
+                }
+            };
+
+            const loadImageFile = () => {
+                requestSender.sendStatefulCompute(
+                    appState,
+                    function(response) {
+                        uris = response.uris;
+                        setImageFromUriIndex(0);
+                        loading = false;
+                    },
+                    {
+                        method: 'sample_images',
+                        args: {
+                            filename: 'sample'
+                        }
+                    }
+                );
+            };
+
+            loadImageFile();
+        }
+    };
+});
+
```

Comment:
Creating a new ML simulation in a fresh sirepo install is causing this error:
angular.js:15567 TypeError: Cannot read properties of undefined (reading '0')
    at setImageFromUriIndex (activait.js?20221017.155522:1164:60)

I think the imagePreviewPanel needs to be created only if the Data Source is images and an appropriate file has been added to the simulation.